File: Challenges/Java/FileNaming.java
public final class FileNaming {

	String[] fileNaming(String[] names) {
	    List<String> list = new ArrayList();
	    for(int i = 0; i < names.length; i++) {
	        if(!list.contains(names[i])) {
	        	list.add(names[i]);
	        	continue;
	        }
	        String s = names[i];
            for(int n = 1; list.contains(s); n++) {
                s = names[i]+"("+n+")";
            }
            list.add(s);
	    }
	    return list.toArray(new String[list.size()]);
	}
}

File: Challenges/Javascript/fileNaming.js
function fileNaming(names) {
    let list = [];
    for(let i in names) {
        if(!list.includes(names[i])) {
            list.push(names[i]);
            continue;
        }
        let s = names[i];
        for(let n = 1; list.includes(s); n++) {
            s = names[i]+"("+n+")";
        }
        list.push(s);
    }
    return list;
}